src/GraphQL/Resolvers/MemberResolver.cs

```diff
@@ -0,0 +1,168 @@
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace GraphQL.Resolvers
+{
+    /// <summary>
+    /// A precompiled field resolver for a specific <see cref="MethodInfo"/>, <see cref="PropertyInfo"/> or <see cref="FieldInfo"/>.
+    /// Returns the specified field or property, or for methods, calls the specified method (with the specified arguments)
+    /// and returns the value of the method.
+    /// </summary>
+    internal class MemberResolver : IFieldResolver
+    {
+        private readonly Func<IResolveFieldContext, object?> _resolver;
+
+        /// <summary>
+        /// Initializes an instance for the specified field, using a default instance expression of:
+        /// <code>context =&gt; (TSourceType)context.Source</code>
+        /// </summary>
+        public MemberResolver(FieldInfo fieldInfo)
+            : this(fieldInfo, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes an instance for the specified field, using the specified instance expression to access the instance of the field.
+        /// If <paramref name="instanceExpression"/> is <see langword="null"/> then a default instance expression is used as follows:
```

Comment:
> then a default instance expression is used as follows: <code>context =&gt; (TSourceType)context.Source</code>

Is such an expression used by default makes sense? Does it make any sense to use any default value? Is it worth demanding it always explicitly?

Reply:
Yes, I expect those constructors can be removed.  Although `ExpressionFieldResolver` and `NameFieldResolver` need to build a 'default instance expression of `context => context.Source`' in PR #2907, it does not do so with those constructors.  So I can remove them.

Reply:
I would not say at all that ExpressionFieldResolver and NameFieldResolver build some "default" expressions. They build concrete expressions like `context => context.Source.MyProperty` and user can not configure those expressions. No defaults.

Reply:
Done
- Excess constructors removed in `MemberResolver` and `EventStreamMethodResolver`
- Instance expression is not nullable for other constructors
- Constructor parameters reordered more appropriately considering that instance expression is not optional
- Instance expression is not nullable in `AutoRegisteringHelper.BuildFieldResolver`
- Xml comments updated.